### PR TITLE
[IMP] account: Update chart_template more testing

### PR DESCRIPTION
**Pull request by clbr-odoo**
Add a test that checks that a chart_template reload when data didn't changed won't delete or create any records.

Test added retroactively for this fix: https://github.com/odoo/odoo/pull/126830

### CODE REVIEW

File: addons/account/tests/test_chart_template.py [hunk 1]
```diff
@@ -363,3 +382,27 @@ def local_get_data(self, template_code):
...
+
+    def test_update_reload_no_new_data(self):
+        """ Tests that the reload does nothing when data are left unchanged.
+        Tested models: account.group, account.account, account.tax.group, account.tax, account.journal,
+        account.reconcile.model, account.fiscal.position, account.fiscal.position.tax, account.tax.repartition.line,
+        account.account.tag.
+        """
+        def get_domain(model):
+            if model == 'account.account.tag':
+                return 'country_id', '=', self.company_1.country_id.id
```
Comment (william-andre) [inline addons/account/tests/test_chart_template.py:394]: This is returning one leaf, not a domain like the function name suggests
```suggestion
                return [('country_id', '=', self.company_1.country_id.id)]
```

File: addons/account/tests/test_chart_template.py [hunk 2]
```diff
@@ -363,3 +382,27 @@ def local_get_data(self, template_code):
         self.assertFalse(parent_tax.active, "The parent tax should be inactive.")
         self.assertEqual(len(children_taxes), 2, "Two children should have been created, even if they are inactive.")
         self.assertEqual(children_taxes.mapped('active'), [False] * 2, "Children taxes should be inactive.")
+
+    def test_update_reload_no_new_data(self):
+        """ Tests that the reload does nothing when data are left unchanged.
```
Comment (william-andre) [inline addons/account/tests/test_chart_template.py:387]: As I understand it, you are only checking the ids of the records, not the data itself?

Comment (clbr-odoo) [reply addons/account/tests/test_chart_template.py:387]: Yes, I can add  `assertRecordValues()` but it implies few more loops.

**Review by william-andre — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**